### PR TITLE
Fixes #619 - Updated PR Template + Misc Docs Fixes

### LINKED ISSUE #619: updated PR template based on v3 changes

### PR DESCRIPTION
**Pull request by m0mosenpai**
# Description

- [X] Updated PR Template
- [x] Added missing info in docs

## Related issues
#619 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue).

# Checklist

- [X] The tests gave 0 errors.
- [X] `Black` gave 0 errors.
- [X] `Flake` gave 0 errors.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -16,15 +16,19 @@ Please delete options that are not relevant.
 # Checklist
 
 - [ ] The pull request is for the branch develop
-- [ ] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
-- [ ] If I added external libraries/packages that use restrictive licenses, please add them in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section
-- [ ] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the docs: [Installation](./Installation.md)
+- [ ] A new analyzer was added, in which case:
```
Comment (sp35) [inline .github/pull_request_template.md:19]: or connector

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -16,15 +16,19 @@ Please delete options that are not relevant.
...
+- [ ] A new analyzer or connector was added, in which case:
+    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
+    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (incase the analyzer provides additional optional configuration).
+    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
+    - [ ] If the analyzer requires mocked testing, `_monkeypatch()` was used in the analyzer's class to apply the necessary decorators.
+    - [ ] If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
```
Comment (sp35) [inline .github/pull_request_template.md:24]: or Connectors

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -16,15 +16,19 @@ Please delete options that are not relevant.
...
-- [ ] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the docs: [Installation](./Installation.md)
+- [ ] A new analyzer or connector was added, in which case:
+    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
+    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (incase the analyzer provides additional optional configuration).
+    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
+    - [ ] If the analyzer requires mocked testing, `_monkeypatch()` was used in the analyzer's class to apply the necessary decorators.
```
Comment (sp35) [inline .github/pull_request_template.md:23]: /connector

**Comment by Eshaan7** [comment]
pls fix the merge conflicts, thank you.

**Pull request merged**